examples/advanced-tensorflow/README.md

```diff
@@ -1,79 +1,94 @@
 
-Write the command below in your terminal to install the dependencies according to the configuration file requirements.txt.
+![](_static/wandb_plots.png)
 
-```shell
-pip install -r requirements.txt
+### Run with the Simulation Engine
+
+With default parameters, 25% of the total 50 nodes (see `num-supernodes` in `pyproject.toml`) will be sampled for `fit` and 50% for an `evaluate` round. By default `ClientApp` objects will run on CPU.
+
+> \[!TIP\]
+> To run your `ClientApps` on GPU or to adjust the degree or parallelism of your simulation, edit the `[tool.flwr.federations.local-simulation]` section in the `pyproject.tom`.
+
+```bash
+flwr run .
+
+# To disable W&B
+flwr run . --run-config use-wandb=false
 ```
 
-## Run Federated Learning with TensorFlow/Keras and Flower
+> \[!WARNING\]
+> By default TensorFlow process that use GPU will try to pre-allocate the entiery available VRAM. This is undesirable for simulations where we want the GPU to be shared among several `ClientApp` instances. Enable the [GPU memory growth](https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth) by setting the `TF_FORCE_GPU_ALLOW_GROWTH` environment variable to ensure processes only make use of the VRAM they need.
 
-The included `run.sh` will call a script to generate certificates (which will be used by server and clients), start the Flower server (using `server.py`), sleep for 10 seconds to ensure the server is up, and then start 10 Flower clients (using `client.py`). You can simply start everything in a terminal as follows:
+You can run the app using another federation (see `pyproject.toml`). For example, if you have a GPU available, select the `local-sim-gpu` federation:
 
-```shell
-# Once you have activated your environment
-./run.sh
+```bash
+export TF_FORCE_GPU_ALLOW_GROWTH="true"
+flwr run . local-sim-gpu
 ```
 
-The `run.sh` script starts processes in the background so that you don't have to open eleven terminal windows. If you experiment with the code example and something goes wrong, simply using `CTRL + C` on Linux (or `CMD + C` on macOS) wouldn't normally kill all these processes, which is why the script ends with `trap "trap - SIGTERM && kill -- -$$" SIGINT SIGTERM EXIT` and `wait`. This simply allows you to stop the experiment using `CTRL + C` (or `CMD + C`). If you change the script and anything goes wrong you can still use `killall python` (or `killall python3`) to kill all background processes (or a more specific command if you have other Python processes running that you don't want to kill).
+You can also override some of the settings for your `ClientApp` and `ServerApp` defined in `pyproject.toml`. For example:
 
-By default `run.sh` uses only a subset of the data. To use the full data, remove the `--toy` argument or set it to False.
+```bash
+flwr run . --run-config "num-server-rounds=5 fraction-fit=0.5"
```

Comment:
```suggestion
flwr run . --run-config "num-server-rounds=10 fraction-fit=0.5"
```

Reply:
The default is num-server-rounds=5, so better to override this with a different value.